internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -169,6 +171,28 @@ func (r *AgentServiceConfigReconciler) Reconcile(origCtx context.Context, req ct
 	return ctrl.Result{}, r.Status().Update(ctx, instance)
 }
 
+func (r *AgentServiceConfigReconciler) ensureServiceMonitor(ctx context.Context, log logrus.FieldLogger, instance *aiv1beta1.AgentServiceConfig) error {
```

Comment:
Add this to the reconcile loop?

Reply:
:facepalm:  I laughed out loud! 

I'm getting back to this now and remembering that I didn't seem to see the `ServiceMonitor` being created, which is what I wanted to debug today (and why this was a draft).... now we both know why it was not being created :stuck_out_tongue: